### ISSUE TITLE
[Feature requset] River support

### ISSUE THREAD

**Issue by Sunderland93** [nwg-piotr/nwg-displays#30]
https://github.com/riverwm/river

**Comment by nwg-piotr** [comment]
Basically nwg-shell and its components are for sway and Hyprland. By adding support for other window managers (such as DWL tags in nwg-panel), I know that I won't use it myself, so the option may be poorly tested and not further developed. At the moment the answer is no. Perhaps on long winter evenings.

**Comment by cvoges12** [comment]
@nwg-piotr I would be happy to maintain and test for riverwm support.

**Comment by nwg-piotr** [comment]
@cvoges12 - sure, go ahead.

**Comment by cvoges12** [comment]
I'm going to notate all the places where sway / hyprland dependent code is taking place and start adding the riverwm equivalents.

So far I've found:
- getting the SWAYSOCK and/or HYPRLAND_INSTANCE_SIGNATURE
- getting the path of the .config directory for sway and hyprland
- creating the corresponding outputs/monitors and workspaces files
- list of active outputs (from sway tree)

I'll keep listing as I go through. I've noticed that there are some generically named functions and such that are only being used for sway. I'll refactor and rename these before I continue to ease the process. I'll send a PR regarding this change separately since it should probably be done anyways regardless if the changes for this issue get merged or not.

**Comment by nwg-piotr** [comment]
Keep in mind that I'm only going to test your changes on sway and Hyprland. River-related stuff remains your responsibility.

**Comment by cvoges12** [comment]
Yep, I'm fully aware. Right now I'm just trying to get a sense of the project and where all the compositor specific code is.
